config/crds/ibmcloud_v1alpha1_ibmcloudmachineproviderspecspec.yaml

```diff
@@ -0,0 +1,53 @@
+apiVersion: apiextensions.k8s.io/v1beta1
```

Comment:
should not this file name be `ibmcloud_v1alpha1_ibmcloudmachineproviderspec.yaml`?

BTW: How did you generate this file?

Reply:
make file generate them , this is auto generated , I think your kubebuilder might have some issue so it lead to weird thing here like statusstatus and specspec definition
I made some changes according to openstack provider

# Generate manifests e.g. CRD, RBAC etc.
manifests:
        go run vendor/sigs.k8s.io/controller-tools/cmd/controller-gen/main.go crd
        kustomize build config/default/ > provider-components.yaml
        echo "---" >> provider-components.yaml
        kustomize build vendor/sigs.k8s.io/cluster-api/config/default/ >> provider-components.yaml